docs/install/ubuntu_setup.md

```diff
@@ -140,9 +140,30 @@ On Ubuntu versions 16.04 or later, you need the following dependencies:
 **Step 1:** Install build tools and git.
 ```bash
     sudo apt-get update
-    sudo apt-get install -y build-essential git
+    sudo apt-get install -y build-essential git ninja-build ccache
 ```
 
+**For Ubuntu 18.04 and CUDA builds you need to update CMake**
+
+```bash
+#!/usr/bin/env bash
+set -exuo pipefail
+sudo apt remove --purge --auto-remove cmake
+
+# Update CMAKE for correct cuda autotedetection: https://github.com/clab/dynet/issues/1457
+version=3.14
```

Comment:
Putting in version number in the docs makes them go stale pretty fast. Is there a way to consolidate versions in a file so it can be managed there instead of sprinkled throughout code blocks in the docs?

Reply:
I think this is fine. This is the minimum version that works with ubuntu 18.04 I think is ok to put it there, and people can update it. If you put an indirect reference becomes more messy to maintain.